poutyne/framework/model.py

```diff
@@ -428,23 +442,27 @@ def predict_on_batch(self, x):
         tensors are converted into Numpy arrays.
 
         Args:
-            x (Union[Tensor, np.ndarray]): Batch for which to predict.
+            x (Union[Tensor, np.ndarray, tuple, list]): Batch for which to predict.
+                If ``x`` is a tuple or list, the tuple or list content must be of Union[Tensor, np.ndarray].
 
         Returns:
             The predictions with tensors converted into Numpy arrays.
         """
+        x = self._format_input(x)
+
         self.model.eval()
         with torch.no_grad():
             x = self._process_input(x)
```

Comment:
Remove 
```
x = self._format_input(x)
```
above and do as in predict_generator:
```suggestion
             x = self._process_input(self._format_input(x))
```